tests/common/common.h

```diff
@@ -753,6 +753,56 @@ inline bool kernel_supports_wg_size(sycl_cts::util::logger& log,
     }                                                                    \
   }
 
+/** \brief Tests the result of using unary operator op with operand val,
+ *         while storing the results in res. */
+#define UNARY_INDEX_KERNEL_TEST(op, val, res)                   \
+  do {                                                          \
+    val = op res;                                               \
```

Comment:
Good catch, this is a mistake. It should have been `res = op val`. I've updated the variables and comments for these macros so that it is more clear what is happening.

There is no test for `res = val op` because this macro is for unary plus and minus. The prefix and postfix tests have separate macros.